android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginViewModel.kt

```diff
@@ -91,11 +111,41 @@ constructor(
       }
     }
 
-  private fun storeUserPreferences(responseBody: ResponseBody) {
-    val responseBodyString = responseBody.string()
-    Timber.d(responseBodyString)
-    val userResponse = responseBodyString.decodeJson<UserInfo>()
-    sharedPreferences.write(USER_INFO_SHARED_PREFERENCE_KEY, userResponse.encodeJson())
+  private fun storeUserPreferences(userInfo: UserInfo) {
+    sharedPreferences.write(USER_INFO_SHARED_PREFERENCE_KEY, userInfo.encodeJson())
+  }
+
+  fun callPractitionerDetails(userResponse: UserInfo) {
+    runBlocking {
+      val bundle = accountAuthenticator.getPractitionerDetails(userResponse.sub!!)
+      val practitionerDetails = bundle.entry[0].resource as PractitionerDetails
```

Comment:
what if we do  not have this

Reply:
will add a check to if bundle.hasEntry().